meta-ostro/classes/ostro-image.bbclass

```diff
@@ -10,121 +10,245 @@ IMAGE_INSTALL = " \
                 ${OSTRO_IMAGE_EXTRA_INSTALL} \
 		"
 
+# In Ostro OS, /bin/sh is always dash, even if bash is installed for
+# interactive use.
+IMAGE_INSTALL += "dash"
+
+# Certain keywords, like "iotivity" are used in different contexts:
+# - as image feature name
+# - as bundle name
+# - optional: as additional image suffix
+#
+# While this keeps the names short, it can also be a bit confusing and
+# makes some of the definitions below look redundant. They are needed,
+# though, because the naming convention could also be different.
+
+
 # Image features sometimes affect image building (for example,
 # ima enables image signing) and/or adds certain packages
 # via FEATURE_PACKAGES.
 #
-# TODO: document IoT specific image feature somewhere. Here?
-IMAGE_FEATURES[validitems] += " \
-    app-privileges \
-    autologin \
+# This is the list of image features which merely add packages.
+# This list also gets turned into the default set of swupd bundles.
+OSTRO_IMAGE_PKG_FEATURES = " \
     can \
     connectivity \
     devkit \
-    ima \
     iotivity \
     java-jdk \
     node-runtime \
     nodejs-runtime-tools \
     python-runtime \
     qatests \
-    smack \
+    ssh-server \
     soletta \
     soletta-tools \
-    swupd \
+    tools-debug \
     tools-develop \
+    tools-interactive \
 "
 
-# These features come from base recipes, but are not added to
-# IMAGE_FEATURES[validitems]. Should better be fixed there.
+# Here is the complete list of image features, also including
+# those that modify the image configuration.
+#
+# TODO: document all relevant (not just IoT) image features
+# in building-images.rst.
+#
+# swupd = install swupd client and enabled generation of swupd bundles
 IMAGE_FEATURES[validitems] += " \
-    ptest-pkgs \
-    ssh-server-openssh \
-    tools-debug \
-    tools-profile \
+    app-privileges \
+    autologin \
```

Comment:
Let's see how that "shell application" goes. Core OS developers will always need the ability to investigate things in the core OS; if the "shell application" can provide that, then we can remove this direct login.